### Теория вероятности-6 [Домашнее задание]

**Примечание:**

> Пожалуйста, не удаляйте комментарии или строки кода.

> Пожалуйста, покажите ваше решение, написав процедуру расчета.

**Например:**

Если вы подбросите монету, какова вероятность получить Tails?

**Ваше решение должно выглядеть так:**

```
p_tails = 1 / (2**1) # Хороший пример решения
# p_tails = 0.5 # Плохой пример решения
```

1. Вы работаете в медицинском исследовательском центре. Был разработан новый тест на $HIV$. Вас просят оценить новый тест, хорошо ли он работает на основе ваших знаний по теории вероятностей. Вам даны результаты эксперимента по новому тесту на $HIV$.
    
    Условные обозначения:
    
    * $HIV^+$ - количество людей, инфицированных $HIV$.
    
    * $HIV^-$ - количество людей, НЕ инфицированных $HIV$.
    
    * $Test^+$ - количество людей, у которых тест показал положительный результат на $HIV$.
    
    * $Test^-$ - количество людей, у которых тест показал отрицательный результат на $HIV$.


||$HIV^+$|$HIV^-$|Total|
|:-:|:-:|:-:|:-:|
|$Test^+$|10 |9    |19   |
|$Test^-$|7  |123  |130  |
|Total   |17 |132  |149  |


На основании предоставленных вам данных, пожалуйста, рассчитайте следующие вероятности::

* $P(Test^+)$, и присвойте результат переменной ```p_tp```.
* $P(Test^-)$, и присвойте результат переменной ```p_tn```.
* $P(HIV^+)$, и присвойте результат переменной ```p_hp```.
* $P(HIV^-)$, и присвойте результат переменной ```p_hn```.
* $P(Test^+|HIV^+)$, и присвойте результат переменной ```p_tp_hp```.
* $P(Test^-|HIV^+)$, и присвойте результат переменной ```p_tn_hp```.
* $P(Test^+|HIV^-)$, и присвойте результат переменной ```p_tp_hn```.
* $P(Test^-|HIV^-)$, и присвойте результат переменной ```p_tn_hn```.

In [7]:
#Начало вашего кода
p_tp = 19 / 149
p_tn = 130 / 149
p_hp = 17 / 149
p_hn = 132 / 149
p_tp_hp = 10/17
p_tn_hp = 7 / 17
p_tp_hn = 9 / 132
p_tn_hn = 123 / 132
#Конец вашего кода

print(f"{p_tp*100:.2f}%")
print(f"{p_tn*100:.2f}%")
print(f"{p_hp*100:.2f}%")
print(f"{p_hn*100:.2f}%")
print(f"{p_tp_hp*100:.2f}%")
print(f"{p_tn_hp*100:.2f}%")
print(f"{p_tp_hn*100:.2f}%")
print(f"{p_tn_hn*100:.2f}%")

12.75%
87.25%
11.41%
88.59%
58.82%
41.18%
6.82%
93.18%


### Ожидаемые результаты:

```
12.75%
87.25%
11.41%
88.59%
58.82%
41.18%
6.82%
93.18%
```

2. Этот вопрос связан с вопросом 1. 
    На основании данных, приведенных в таблице выше, рассчитайте следующие вероятности:

    * $P_{table}(HIV^+|Test^+)$, и присвойте результат переменной ```p_t_hp_tp```.
    * $P_{table}(HIV^+|Test^-)$, и присвойте результат переменной ```p_t_hp_np```.

    Основываясь на теории вероятностей вашего знания и используя известные вам формулы, рассчитайте следующие вероятности:

    * $P_{formula}(HIV^+|Test^+)$, и присвойте результат переменной ```p_f_hp_tp```.
    * $P_{formula}(HIV^+|Test^-)$, и присвойте результат переменной ```p_f_hp_np```.


In [20]:
#Начало вашего кода
p_t_hp_tp = 10 / 19
p_t_hp_tn = 7 / 130
p_f_hp_tp = (p_hp * p_tp_hp) / ((p_hp *  p_tp_hp) + (p_hn * p_tp_hn))
p_f_hp_tn = (p_hp * p_tn_hp) / ((p_hp * p_tn_hp) + (p_hn * p_tn_hn))
#Конец вашего кода
print("На основании данных:")
print(f"{p_t_hp_tp*100:.2f}%")
print(f"{p_t_hp_tn*100:.2f}%")

print("Используя формулы:")
print(f"{p_f_hp_tp*100:.2f}%")
print(f"{p_f_hp_tn*100:.2f}%")


На основании данных:
52.63%
5.38%
Используя формулы:
52.63%
5.38%


### Ожидаемые результаты:

```
На основании данных:
52.63%
5.38%
Используя формулы:
52.63%
5.38%
```

3. Вы вычислили следующие вероятности в предыдущих вопросах: $P(HIV^+)$, $P(HIV^+|Test^+)$, $P(HIV^+|Test^-)$. 
Определите, какие из них являются априорной вероятностью и апостериорной вероятностью. Назначьте ```"prior"```, если вероятность является априорной, или ```"posterior"``` в противном случае для следующих переменных:

    * ```p_hp_type```

    * ```p_hp_tp_type```

    * ```p_hp_tn_type```


In [22]:
#Начало вашего кода
p_hp_type = 'prior'
p_hp_tp_type = 'posterior'
p_hp_tn_type = 'posterior'

#Конец вашего кода
print(p_hp_type)
print(p_hp_tp_type)
print(p_hp_tn_type)

prior
posterior
posterior


### Ожидаемые результаты:

```
prior
posterior
posterior
```

Пожалуйста, попробуйте объяснить априорную и апостериорную вероятности, которые вы получили из предыдущих вопросов.

**Начало вашего ответа:**

Априорная вероятность - это та вероятность, которую мы можем вычислить на основе уже имеющихся данных, до появления новых. После того как у нас появится новая информация и мы вычислим новую вероятность, она станет апостериорной, вероятность на основе обновленных данных. По имеющейся таблице мы знаем 𝑃(𝐻𝐼𝑉+) - 11.41%, это априорная вероятность. После того как мы получим результаты теста вероятность станет более точной - 𝑃(𝐻𝐼𝑉+|𝑇𝑒𝑠𝑡+) - 52.63% или 𝑃(𝐻𝐼𝑉+|𝑇𝑒𝑠𝑡−) - 5.38% , это апостериорные вероятности.


**Конец вашего ответа**

4. Во многих спам письмах вы можете увидеть слово ```"urgent"```. Однако некоторые электронные письма не являются спамом, даже если они содержат слово «urgent». Допустим, мы знаем следующие вероятности:

    * Вероятность получить **спам письмо** по электронной почте: ```0.27```
    
    * Вероятность того, что электронное **спам** письмо содержит слово ```"urgent"```: ```0.46```
    
    * Вероятность того, что электронное **НЕ спам** письмо содержит слово ```"urgent"```: ```0.19```
    
    

Пожалуйста, рассчитайте следующие вероятности:

* $P("spam")$, и присвойте результат переменной ```p_spam```.
* $P("non-spam")$, и присвойте результат переменной ```p_non_spam```.
* $P("urgent"|"spam")$, и присвойте результат переменной ```p_urgent_given_spam```.
* $P("urgent"|"non-spam")$, и присвойте результат переменной ```p_urgent_given_non_spam```.
* $P("urgent"\cap "spam")$, и присвойте результат переменной ```p_urgent_and_spam```.
* $P("urgent"\cap "non-spam")$, и присвойте результат переменной ```p_urgent_and_non_spam```.
* $P("urgent")$, и присвойте результат переменной ```p_urgent```.
* $P("spam"|"urgent")$, и присвойте результат переменной ```p_spam_given_urgent```.
* $P("non-spam"|"urgent")$, и присвойте результат переменной ```p_non_spam_given_urgent```.


In [5]:
#Начало вашего кода
p_spam = 0.27
p_non_spam = 1 - p_spam
p_urgent_given_spam = 0.46
p_urgent_given_non_spam = 0.19
p_urgent_and_spam = p_spam * p_urgent_given_spam
p_urgent_and_non_spam = p_non_spam * p_urgent_given_non_spam
p_urgent = p_spam * p_urgent_given_spam + p_non_spam * p_urgent_given_non_spam
p_spam_given_urgent = p_urgent_and_spam / p_urgent
p_non_spam_given_urgent = p_urgent_and_non_spam / p_urgent
#Конец вашего кода

print(f"{p_spam*100:.2f}%")
print(f"{p_non_spam*100:.2f}%")
print(f"{p_urgent_given_spam*100:.2f}%")
print(f"{p_urgent_given_non_spam*100:.2f}%")
print(f"{p_urgent_and_spam*100:.2f}%")
print(f"{p_urgent_and_non_spam*100:.2f}%")
print(f"{p_urgent*100:.2f}%")
print(f"{p_spam_given_urgent*100:.2f}%")
print(f"{p_non_spam_given_urgent*100:.2f}%")

27.00%
73.00%
46.00%
19.00%
12.42%
13.87%
26.29%
47.24%
52.76%


### Ожидаемые результаты:

```
27.00%
73.00%
46.00%
19.00%
12.42%
13.87%
26.29%
47.24%
52.76%
```

5. Вы вычислили следующие вероятности в предыдущем вопросе: $P("spam")$, $P("spam"|"urgent")$.

    * Присвойте априорную вероятность получить спам по электронной почте переменной: ```prior```.
    
    * Присвойте апостериорную вероятность получить спам по электронной почте переменной: ```posterior```.
    
    * Рассчитайте соотношение между апостериорной и априорной вероятностью - вам нужно разделить апостериорную вероятность на априорную вероятность. Присвойте ответ переменной: ```ratio```.

In [6]:
#Начало вашего кода
prior = p_spam
posterior = p_spam_given_urgent
ratio = posterior / prior
#Конец вашего кода
print(f"{prior*100:.2f}%")
print(f"{posterior*100:.2f}%")
print(f"{ratio:.2f}")

27.00%
47.24%
1.75


### Ожидаемые результаты:

```
27.00%
47.24%
1.75
```

Пожалуйста, попробуйте объяснить априорную, апостериорную вероятности и их соотношение , которые вы получили.

**Начало вашего ответа:**

Априорная вероятность - это вероятность получить спам, она нам была изначально. Если мы знаем что сообщение имеет фразу "urgent", то вороятность что это сообщение спам увеличивается, это  апостериорная вероятность. Соотношение между ними 1.75. Это озночает, что апостериорная вероятность на 175% более точная чем априорная вероятность

**Конец вашего ответа**

6. Если у нас есть одна стандартная колода из 52 карт, какие из следующих пар событий являются взаимоисключающими, а какие нет? Присвойте переменным ```"exclusive"``` - взаимоисключающие, ```"non-exclusive"``` - НЕ взаимоисключающие.

    a. Вытащить карту красной масти и вытащить короля? Присвойте результат переменной ```answer_6a```.

    b. Вытащить карту черной масти и вытащить валет? Присвойте результат переменной ```answer_6b```.

    c. Вытащить карту красной масти и вытащить карту масти пики? Присвойте результат переменной ```answer_6c```.

    d. Вытащить карту черной масти и вытащить туз буби? Присвойте результат переменной ```answer_6d```.

In [9]:
#Начало вашего кода
answer_6a = "non-exclusive"
answer_6b = "non-exclusive"
answer_6c = "exclusive"
answer_6d = "exclusive"
#Конец вашего кода
print(answer_6a)
print(answer_6b)
print(answer_6c)
print(answer_6d)

non-exclusive
non-exclusive
exclusive
exclusive


### Ожидаемые результаты:

```
non-exclusive
non-exclusive
exclusive
exclusive
```

7. Если у нас есть одна стандартная колода из 52 карт, какие из следующих пар событий являются независимыми, а какие нет? Присвойте переменным ```"independent"``` - независимыe, ```"dependent"``` - зависимыe.

    a. Вытащить карту красной масти и вытащить короля? Присвойте результат переменной ```answer_7a```.

    b. Вытащить карту черной масти и вытащить валет? Присвойте результат переменной ```answer_7b```.

    c. Вытащить карту красной масти и вытащить карту масти пики? Присвойте результат переменной ```answer_7c```.

    d. Вытащить карту черной масти и вытащить туз буби? Присвойте результат переменной ```answer_7d```.

In [1]:
#Начало вашего кода
answer_7a = "independent"
answer_7b = "independent"
answer_7c = "dependent"
answer_7d = "dependent"

#Конец вашего кода
print(answer_7a)
print(answer_7b)
print(answer_7c)
print(answer_7d)

independent
independent
dependent
dependent


### Ожидаемые результаты:

```
independent
independent
dependent
dependent
```

### Отличная работа